### PR TITLE
fix(openai): fall back to reasoning_content when content is empty

### PR DESCRIPTION
**Pull request by xiaoyesoso**
Reasoning models served via OpenAI-compatible endpoints (Xinference, vLLM, etc.) -- e.g. qwen3-thinking, deepseek-r1 -- can return content='' with the partial answer in reasoning_content when the generation is truncated by max_tokens (finish_reason='length'). The OpenAI adapter previously raised 'LLM returned empty content and no tool calls' in that case, which made it impossible to validate or activate such a model from the UI's connect-test flow.

* chat() and vision_chat() now read reasoning_content once and, when content is empty/None and there are no tool calls but reasoning_content is populated, surface the reasoning text as content (with reasoning_content/reasoning fields preserved on the result). The hard error is still raised when both content and reasoning_content are empty.

* Mirrors the Xinference adapter behaviour landed in #625, so both adapters handle truncated reasoning responses consistently.

* test_none_content_response and test_empty_content_response now explicitly pin reasoning_content=None so the empty-response error path keeps being exercised; new test_empty_content_falls_back_to_reasoning_content covers the recovery path.

### CODE REVIEW

File: src/xagent/core/model/chat/basic/openai.py [hunk 1]
```diff
@@ -268,9 +268,33 @@ def _process_response(resp: Any) -> Dict[str, Any]:
...
+                # ``content=""`` while ``reasoning_content`` carries the
+                # partial answer when the generation is truncated by
+                # ``max_tokens`` (``finish_reason="length"``) before the
+                # final answer is produced. Surface the reasoning text as
+                # content so callers (notably the model connection test) do
+                # not treat a truncated-but-otherwise-healthy response as
+                # invalid. Mirror the ``content`` whitespace check so a
+                # reasoning trace that is purely whitespace still falls
+                # through to the empty-response error.
+                if reasoning_content and reasoning_content.strip():
```
Comment (qinxuye) [inline src/xagent/core/model/chat/basic/openai.py]: The PR description and test case scope this fallback to `finish_reason="length"` truncation, but the implementation promotes `reasoning_content` whenever `content` is empty. `content` is consumed downstream as the normal assistant output, so a provider response with `finish_reason="stop"`, empty final content, and a populated reasoning trace would now be treated as a successful answer instead of surfacing that the model never produced final content. Please gate this fallback on the choice being truncated, e.g. `finish_reason == "length"`, and add a regression test where `finish_reason="stop"` with empty content and non-empty reasoning still raises. The same guard should be applied to the `vision_chat` path below.

**Review by qinxuye — COMMENTED** [review]
One issue found in the reasoning_content fallback scope.

File: src/xagent/core/model/chat/basic/openai.py (hunk 1 above)
Comment (xiaoyesoso) [reply src/xagent/core/model/chat/basic/openai.py]: Addressed in 9c12fc2:

- chat() and vision_chat() now read finish_reason from the choice and gate the
  reasoning_content fallback strictly on finish_reason == "length". Any other
  terminal reason (stop / content_filter / None) falls through to the existing
  empty-response RuntimeError.

- New test_finish_reason_stop_with_only_reasoning_still_raises pins the contract
  you described: finish_reason="stop" with empty content and a populated
  reasoning trace must surface as the empty-response error rather than be
  promoted to content.

The existing test_empty_content_falls_back_to_reasoning_content still uses
finish_reason="length" and continues to verify the truncated-recovery path.

File: src/xagent/core/model/chat/basic/openai.py [hunk 2]
```diff
@@ -268,9 +268,45 @@ def _process_response(resp: Any) -> Dict[str, Any]:
...
+                # hide the bug.
+                if (
+                    finish_reason == "length"
+                    and reasoning_content
+                    and reasoning_content.strip()
+                ):
+                    return {
+                        "type": "text",
+                        "content": reasoning_content,
+                        "reasoning_content": reasoning_content,
```
Comment (rogercloud) [inline src/xagent/core/model/chat/basic/openai.py:306]: Nit: `reasoning_content` and `reasoning` are both set to the same value on every response dict (here, line 321, and in `vision_chat` lines 676/691). A brief comment noting which downstream consumers expect which key would help future maintainers — e.g. `# raw field from provider` vs `# canonical key for pattern layer`.

File: tests/core/model/chat/basic/test_openai.py [hunk 3]
```diff
@@ -406,6 +412,133 @@ async def test_empty_content_response(self, openai_llm_config, mocker):
         ):
             await llm.chat([{"role": "user", "content": "Hello"}])
```
Comment (rogercloud) [inline tests/core/model/chat/basic/test_openai.py:413]: Suggestion: the test coverage is strong for the happy path and rejection cases, but two edge cases in the `finish_reason` gate are not exercised:
- `finish_reason=None` with populated `reasoning_content` (should fall through to error)
- `finish_reason="content_filter"` with populated `reasoning_content` (should also fall through)

Both are handled correctly by the current code (`== "length"` is strict), but explicit tests would make the intent clearer and prevent accidental broadening of the fallback in the future.

File: src/xagent/core/model/chat/basic/openai.py [hunk 4]
```diff
@@ -610,19 +646,51 @@ async def vision_chat(
 
             # Handle text content
             content = message.content
```
Comment (rogercloud) [inline src/xagent/core/model/chat/basic/openai.py:648]: Note: the `vision_chat` function now has the same reasoning fallback logic as `chat()` but the test suite only exercises the `chat()` path. This is a pre-existing coverage gap (the only vision test is `test_vision_chat_preserves_zero_temperature`), not introduced by this PR, but worth noting as the duplicated logic increases the risk of one path drifting.

**Review by rogercloud — APPROVED** [review]
## Review Summary

Clean mirror of #625 for the OpenAI adapter. Same pattern, same quality — reasoning models truncated by `max_tokens` can return `content=""` with the partial answer in `reasoning_content`, and the adapter now surfaces that instead of treating the response as invalid.

### Test Results ✅

```
test_empty_content_falls_back_to_reasoning_content     PASSED
test_whitespace_only_reasoning_content_still_raises    PASSED
test_finish_reason_stop_with_only_reasoning_still_raises PASSED
test_none_content_response (updated)                  PASSED
test_empty_content_response (updated)                 PASSED
```

5/5 relevant tests pass.
